src/util/change-types.ts

```diff
@@ -52,64 +59,70 @@ export const changeStationType = (
 };
 
 /**
- * Change all the stations' type of currentStnType to newStnType in batch.
+ * Change stations' type of currentStnType to newStnType in batch.
  * @param graph Graph.
  * @param currentStnType Current station's type.
  * @param newStnType New station's type.
+ * @param stations Selected stations (undefined for all)
  * @returns Nothing.
  */
 export const changeStationsTypeInBatch = (
     graph: MultiDirectedGraph<NodeAttributes, EdgeAttributes, GraphAttributes>,
-    currentStnType: StationType,
-    newStnType: StationType
+    currentStnType: StationType | 'any',
+    newStnType: StationType,
+    stations?: StnId[]
```

Comment:
Let's do the select-all operation in the caller. This should always be _defined_. It's ok to pass an empty array here but would not be intuitive to do a select-all operation based on whether stations are _defined_. Same for all other change in batch methods.

Reply:
Is `StnId[] | 'any'` ok? An empty array means that the station is not modified.

Reply:
An empty array should mean that no station will be modified. But for all stations, the caller should pass all stations in `stations` instead of `undefined`.